### DISCUSSION TITLE
Working on switching a service from FlaskAPI to FastAPI. The FlaskAPI document uses swag_from to import the swagger UI from a yml file, but I can't find a simple way to do this in FastAPI. Is there a simple way?

### DISCUSSION THREAD

**Discussion by leenatantawy** [fastapi/fastapi#6297] [Questions]
### First Check

- [X] I added a very descriptive title to this issue.
- [X] I used the GitHub search to find a similar issue and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/samuelcolvin/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
#tried the below solution already
from fastapi.openapi.utils import get_openapi

def getrsvp_openapi():
          openapi_data = router.openapi()
    # Change "openapi.json" to desired filename
    with open("get_rsvp_challenge.yml", "w") as file:
        json.dump(openapi_data, file)

#below is the code in flaskapi

from flasgger import swag_from

    @swag_from('folder/filename.yml')
```

### Description

I assumed that there would be a quick replacement for swag_from in fast API but am unable to find one, and most of the documentation I found doesn't show .yml examples.

### Operating System

macOS

### FastAPI Version

0.79.0

### Python Version

Python 3.8.8rc1

**Comment by sparsh820** [comment]
`from fastapi import FastAPI, Path
from pydantic import BaseModel
from typing import List, Dict
from enum import Enum
import yaml

class PaletteEnum(str, Enum):
    all = "all"
    rgb = "rgb"
    cmyk = "cmyk"

class Palette(BaseModel):
    palette_name: List[str]

app = FastAPI()

@app.get("/")
def read_root():
    return {"message": "Welcome to the FastAPI App"}


def get_colors(palette_new: PaletteEnum = Path(..., description="The palette to get colors for: all, rgb, cmyk")):
    """
       {description}
    """
    all_colors = {
        'cmyk': ['cyan', 'magenta', 'yellow', 'black'],
        'rgb': ['red', 'green', 'blue']
    }
    if palette_new == PaletteEnum.all:
        result = all_colors
    else:
        result = {palette_new.value: all_colors.get(palette_new.value)}

    return result
s=''
with open('colors.yml', 'r') as file:
    custom_schema = yaml.load_all(file,yaml.FullLoader)
    s+=''
    for doc in custom_schema:
        print(type(doc))
        s+=str(doc)
print(s)
get_colors.__doc__=get_colors.__doc__.format(description=s)
app.add_api_route("/colors/{palette_new}",get_colors,response_model=Dict[str, List[str]])




def useful():
    """{something}"""

useful.__doc__ = useful.__doc__.format(something="Why do you want this?")
app.add_api_route("/useful", useful)



if __name__ == "__main__":
    import uvicorn
    uvicorn.run(app, host="127.0.0.1", port=8000, log_level="debug")
`
This code can help in generating docstring from yml file . Another approach is to switch to corressponding docstring for yml file.Good Luck!

**Comment by niallmandal** [comment]
okay buddy
